cmd/plume/release.go

```diff
@@ -445,3 +470,84 @@ func doAWS(ctx context.Context, client *http.Client, src *storage.Bucket, spec *
 		}
 	}
 }
+
+func doS3(spec *fcosChannelSpec) {
+	api, err := aws.New(&aws.Options{
+		CredentialsFile: awsCredentialsFile,
+		Profile:         spec.Profile,
+		Region:          spec.Region,
+	})
+	if err != nil {
+		plog.Fatalf("creating aws client: %v", err)
+	}
+
+	// Assumes the bucket layout defined inside of
+	// https://github.com/coreos/fedora-coreos-tracker/issues/189
+	err = api.UpdateBucketObjectsACL(spec.Bucket, filepath.Join("prod", "streams", specChannel, "builds", specVersion), specPolicy)
+	if err != nil {
+		plog.Fatalf("updating object ACLs: %v", err)
+	}
+}
+
+func modifyReleaseMetadataIndex(spec *fcosChannelSpec, commitId string) {
+	api, err := aws.New(&aws.Options{
+		CredentialsFile: awsCredentialsFile,
+		Profile:         spec.Profile,
+		Region:          spec.Region,
+	})
+	if err != nil {
+		plog.Fatalf("creating aws client: %v", err)
+	}
+
+	path := filepath.Join("prod", "streams", specChannel, "releases.json")
+
+	f, err := api.DownloadFile(spec.Bucket, path)
+	if err != nil {
+		plog.Fatalf("downloading release metadata index: %v", err)
+	}
+	defer f.Close()
+
+	data, err := ioutil.ReadAll(f)
+	if err != nil {
+		plog.Fatalf("reading release metadata index: %v", err)
+	}
+
+	var m ReleaseMetadata
+	err = json.Unmarshal(data, &m)
+	if err != nil {
+		plog.Fatalf("unmarshaling release metadata json: %v", err)
+	}
+
+	url, err := url.Parse(fmt.Sprintf("https://%v.s3.amazonaws.com/prod/streams/%v/builds/%v/release.json", spec.Bucket, specChannel, specVersion))
+	if err != nil {
+		plog.Fatalf("creating metadata url: %v", err)
+	}
+
+	newRel := BuildMetadata{
+		CommitHash: specCommitId,
+		Version:    specVersion,
+		Endpoint:   url.String(),
+	}
+
+	for _, rel := range m.Releases {
+		if rel == newRel {
+			// this BuildMetadata already exists, exit
```

Comment:
Should we sanity check here that if it already exists, that it's the last element in the array? Given that we're relying on ordering.

Reply:
I don't like the idea of moving entries in the array around (nor do I really like the fact that something is relying on array ordering but that's another topic), so I guess I can check if the last entry is exactly what we're wanting to post and otherwise just add another new entry (potentially creating a duplicate but making this release the latest ordered one while not retroactively modifying things).